apiserver/facades/agent/instancemutater/shim.go

```diff
@@ -90,6 +90,9 @@ type charmShim struct {
 
 func (s *charmShim) LXDProfile() lxdprofile.Profile {
 	profile := s.Charm.LXDProfile()
+	if profile == nil {
```

Comment:
I think this needs to initialise the maps in Config and Devices, otherwise it's likely to just push the problem further down the code.

Reply:
No, we don't need this.